daemon/k8s/init.go

```diff
@@ -128,6 +128,18 @@ func WaitForNodeInformation(ctx context.Context, log logrus.FieldLogger, localNo
 			logfields.K8sNodeIP:        k8sNodeIP,
 		}).Info("Received own node information from API server")
 
+		// If the host does not have an IPv6 address, return an error
+		if option.Config.EnableIPv6 {
+			if nodeIP6 == nil {
+				log.WithFields(logrus.Fields{
+					logfields.NodeName:         n.Name,
+					logfields.IPAddr + ".ipv4": nodeIP4,
+					logfields.IPAddr + ".ipv6": nodeIP6,
+				}).Fatal("No IPv6 support on node as ipv6 address is nil")
```

Comment:
I don't think you need to `Fatal` here. Using `Error` for logging and then returning an error is enough: the agent will be stopped anyway because of the failed hive start phase.